aspnetcore/fundamentals/app-state/sample/src/WebAppSession/Controllers/HomeController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+using System;
+
+namespace WebAppSession.Controllers
+{
+    #region snippet1
+    public class HomeController : Controller
+    {
+        const string SessionKeyName = "_SessionKeyName";
+        const string SessionKeyYearsMember = "_SessionKeyYearsMember";
+        const string SessionKeyDate = "_SessionKeyDate";
+
+        public IActionResult Index()
+        {
+            // Requires using Microsoft.AspNetCore.Http;
+            HttpContext.Session.SetString(SessionKeyName, "Rick");
+            HttpContext.Session.SetInt32(SessionKeyYearsMember, 3);
+            return RedirectToAction("SessionNameYears");
+        }
+        public IActionResult SessionNameYears()
+        {
+            var name = HttpContext.Session.GetString(SessionKeyName);
+            var yearsMember = HttpContext.Session.GetInt32(SessionKeyYearsMember);
+
+            return Content($"Name: \"{name}\",  Membership years: \"{yearsMember}\"");
+        }
+        #endregion
+
+        #region snippet2
+        public IActionResult SetDate()
+        {
+            HttpContext.Session.Set<DateTime>(SessionKeyDate, DateTime.Now);
```

Comment:
There are lots of different ways to serialize data, we didn't want to be tied to a specific serializer. Something like this would make a good Asp.Net Contrib package though.